examples/manifold/plot_compare_methods.py

```diff
@@ -18,74 +18,192 @@
-Axes3D
+rng = RandomState(0)
+
+n_samples = 1500
+S_points, S_color = datasets.make_s_curve(n_samples, random_state=rng)
+
+# %%
+# Let's look at the original data. Also define some helping
+# functions, which we will use further on.
+
+
+def plot_3d(points, points_color, title):
+    x, y, z = points.T
+
+    fig, ax = plt.subplots(
+        figsize=(6, 6),
+        facecolor="white",
+        tight_layout=True,
+        subplot_kw={"projection": "3d"},
+    )
+    fig.suptitle(title, size=16)
+    col = ax.scatter(x, y, z, c=points_color, s=50, alpha=0.8)
+    ax.view_init(azim=-60, elev=9)
+    ax.xaxis.set_major_locator(ticker.MultipleLocator(1))
+    ax.yaxis.set_major_locator(ticker.MultipleLocator(1))
+    ax.zaxis.set_major_locator(ticker.MultipleLocator(1))
+
+    fig.colorbar(col, ax=ax, orientation="horizontal", shrink=0.6, aspect=60, pad=0.01)
+    plt.show()
+
+
+def plot_2d(points, points_color, title):
+    fig, ax = plt.subplots(figsize=(3, 3), facecolor="white", constrained_layout=True)
+    fig.suptitle(title, size=16)
+    add_2d_scatter(ax, points, points_color)
+    plt.show()
+
+
+def add_2d_scatter(ax, points, points_color, title=None):
+    x, y = points.T
+    ax.scatter(x, y, c=points_color, s=50, alpha=0.8)
+    ax.set_title(title)
+    ax.xaxis.set_major_formatter(ticker.NullFormatter())
+    ax.yaxis.set_major_formatter(ticker.NullFormatter())
+
+
+plot_3d(S_points, S_color, "Original S-curve samples")
+
+# %%
+# Define algorithms for the Manifold learning
+# -------------------------------------------
+#
+# Manifold learning is an approach to non-linear dimensionality reduction.
+# Algorithms for this task are based on the idea that the dimensionality of
+# many data sets is only artificially high.
+#
+# Read more in the :ref:`User Guide <manifold>`.
+
+n_neighbors = 12  # neighborhood which is used to recover the locally linear structure
+n_components = 2  # number of coordinates for the manifold
+
+# %%
+# Locally Linear Embeddings
+# ^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# Locally linear embedding (LLE) can be thought of as a series of local
+# Principal Component Analyses which are globally compared to find the
+# best non-linear embedding.
+# Read more in the :ref:`User Guide <locally_linear_embedding>`.
+
+params = {
+    "n_neighbors": n_neighbors,
+    "n_components": n_components,
+    "eigen_solver": "auto",
+    "random_state": rng,
+}
+
+# --- Points representation
+lle_standart = manifold.LocallyLinearEmbedding(method="standard", **params)
+S_standart = lle_standart.fit_transform(S_points)
+
+lle_ltsa = manifold.LocallyLinearEmbedding(method="ltsa", **params)
+S_ltsa = lle_ltsa.fit_transform(S_points)
+
+lle_hessian = manifold.LocallyLinearEmbedding(method="hessian", **params)
+S_hessian = lle_hessian.fit_transform(S_points)
+
+lle_mod = manifold.LocallyLinearEmbedding(method="modified", modified_tol=0.8, **params)
+S_mod = lle_mod.fit_transform(S_points)
+
+# --- Plot
```

Comment:
Let's create a separate cell for the plotting.

```suggestion
# %%
```

Reply:
Done